pkg/etcd/etcd.go

```diff
@@ -854,3 +854,82 @@ func backupDirWithRetention(dir string, maxBackupRetention int) (string, error)
 	}
 	return backupDir, nil
 }
+
+func GetServerURLFromETCD(ctx context.Context, cfg *config.Control) (string, int, error) {
+	if cfg.Runtime == nil {
+		return "", 0, fmt.Errorf("runtime is not ready yet")
+	}
+	cl, err := getClient(ctx, cfg.Runtime, endpoint)
+	if err != nil {
+		return "", 0, err
+	}
+	etcdResp, err := cl.KV.Get(ctx, serverKey)
+	if err != nil {
+		return "", 0, err
+	}
+
+	if etcdResp.Count < 1 {
+		return "", 0, fmt.Errorf("servers urls are not yet set")
+	}
+	serverURL := strings.TrimSuffix(string(etcdResp.Kvs[0].Value), "\n")
+	serverURL = serverURL[1 : len(serverURL)-1]
+
+	serverAddress := strings.Split(serverURL, ":")
+	port, err := strconv.Atoi(serverAddress[1])
+	if err != nil {
+		return "", 0, fmt.Errorf("port is not set correctly: %v", err)
+	}
+	return serverAddress[0], port, nil
+}
+
+func SetSelfServerURL(ctx context.Context, cfg *config.Control, privateIP string, port int) {
```

Comment:
It appears that every control-plane server will, on startup, update this key to set themselves as the bootstrap endpoint for the control-plane load balancer. Is that intended?

The warnings in this goroutine should include some indication of where they come from, for example:
```
logrus.Warnf("Unable to register control-plane address with etcd: %v", err)
```